### ISSUE TITLE
Java 9 breaks publish feature on Mac

### ISSUE THREAD

**Issue by mgoricki** [OraOpenSource/apex-nitro#225]
### Issue Description
After installing Java 9.0.1 publish feature doesn't work. I didn't try it on Windows.

### APEX Nitro Version
APEX Nitro version 4.0.0-beta.5

### Operating System
macOS Sierra (10.12.6)

### Node Version
v6.11.2

**Comment by vincentmorneau** [comment]
Bug confirmed. I am seeking help from @krisrice here: https://gist.github.com/krisrice/8da5e6cbc07db5becde799742a3c72d2#gistcomment-2247937

**Comment by vincentmorneau** [comment]
Will follow up here https://github.com/vincentmorneau/apex-publish-static-files/issues/11

**Comment by fuzziebrain** [comment]
@mgoricki Change the file extension for the SQLcl binary `sql.exe`, e.g. `sql.ex_`. This should force the OS to use the `sql.bat` instead. Kris is aware, but I think it will be a while before we see a fix. The last I checked, the application used to create the Windows executable doesn't yet support Java 9. Hope that helps.

**Comment by mgoricki** [comment]
@fuzziebrain I'm on macOS. I think this will not work.

**Comment by fuzziebrain** [comment]
@mgoricki My bad. Don't know why but I thought I saw win 10 somewhere.

**Comment by krisrice** [comment]
Something in nashorn's extending a class is acting up.  I'm waiting on an answer from them. There's other ways to do this if I don't hear from them this week.

**Comment by krisrice** [comment]
It's a bug in java9 that worked in java8 and now fixed in java10. I'll have to rewrite this to work in 9.

**Comment by vincentmorneau** [comment]
Thanks @krisrice I appreciate the follow-up. If this gets fixed in java 9.x patches then I think it can wait. 

Looking forward to the rewrite.

**Comment by vincentmorneau** [comment]
Following @krisrice rewrite, I was able to make it work on both Java 8 and 9. Thanks again!

A small change: I recreated the `walkFileTree` function and made it recursive so it looks at a directory tree, not just the files.

Commit here: https://github.com/vincentmorneau/apex-publish-static-files/commit/463cb92d37db2dba0080799ebc3521f438b02ca6

I will close this ticket once a new release is out.
